agent/internal/taskoutput/test_log.go

```diff
@@ -132,39 +132,47 @@ func (h *testLogDirectoryHandler) watch(ctx context.Context) {
 		h.logger.Execution().Critical(recovery.HandlePanicWithError(recover(), nil, "test log directory watcher"))
 	}()
 
+	timer := time.NewTimer(0)
```

Comment:
could a ticker work?

Reply:
I think so, what would be the advantage?

Reply:
I mean, we wouldn't need to `timer.Reset(time.Millisecond)` later.
Actually, on second thought maybe we're better off with the timer. That way if walking takes longer than the timer's period we'll still get a break between walks.